source/appModuleHandler.py

```diff
@@ -362,6 +363,22 @@ def _getExecutableFileInfo(self):
 		fileinfo = getFileVersionInfo(fileName, "ProductName", "ProductVersion")
 		return (fileinfo["ProductName"], fileinfo["ProductVersion"])
 
+	def _getImmersivePackageInfo(self):
+		# Used to obtain full package structure for a hosted app.
+		# The package structure consists of product name, version, architecture, language, and app ID.
+		# This is useful for confirming whether an app is hosted or not despite an app reporting otherwise.
+		# Some apps such as File Explorer says it is an immersive process but error 15700 is shown.
+		# Others such as Store version of Office are not truly hosted apps but are distributed via Store.
+		length = ctypes.c_uint()
+		buf = ctypes.windll.kernel32.GetPackageFullName(self.processHandle, ctypes.byref(length), None)
+		packageFullName = ctypes.create_unicode_buffer(buf)

```

Comment:
The argument to create_unicode_buffer should be length, not buf I believe? 
According to https://docs.microsoft.com/en-us/windows/win32/api/appmodel/nf-appmodel-getpackagefullname
GetPackageFullname returns ERROR_SUCCESS or an error code. The needed length is written to the length parameter.